src/include/migraphx/matcher.hpp

```diff
@@ -857,23 +857,69 @@ auto skip_broadcasts_converts(Ms... ms)
     return skip(name("broadcast", "multibroadcast", "contiguous", "convert"))(ms...);
 }
 
-template <class T>
-inline auto has_value(T x, float tolerance = 1e-6)
+template <class F>
+inline auto literal_value_checker(F f)
 {
     return skip_broadcasts_converts(make_basic_pred_matcher([=](instruction_ref ins) {
         if(ins->name() != "@literal")
             return false;
         auto l = ins->get_literal();
         if(l.empty())
             return false;
+        return f(l);
+    }));
+}
+
+/**
+ * Uses integer multiples of the corresponding floating point epsilon and
+ * compares with abs(y - x) < eps * (atol_mult + rtol_mult * abs(y))
+ */
+template <class T>
+inline auto has_value(T x, std::size_t atol_mult, std::size_t rtol_mult)
```

Comment:
I dont think its a good idea to have an absolute and relative tolerance like this since they are very different range of numbers. The `atol_mult` is a multiple of epsilon so you would expect a large number like 80 or so, but `rtol_mult` is a multiply of the absolute value so it should be a very small number like `1e-6`. You make this same mistake in the gelu matcher as well.